bpf/bpf_lxc.c

```diff
@@ -973,19 +981,19 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 		 */
 		ep = lookup_ip4_endpoint(ip4);
 		if (ep) {
-#ifdef ENABLE_ROUTING
 			if (ep->flags & ENDPOINT_F_HOST) {
+				if (is_defined(ENABLE_ROUTING)) {
 #ifdef HOST_IFINDEX
-				goto to_host;
-#else
-				return DROP_HOST_UNREACHABLE;
+					goto to_host;
 #endif
+					return DROP_HOST_UNREACHABLE;
+				}
+				goto pass_to_stack;
```

Comment:
For `ENDPOINT_F_HOST` and per-EP routes, this jump to `pass_to_stack` means we're bypassing quite a bit of code. Can you explain why [the section right below](https://github.com/cilium/cilium/blob/fcf2b7ad1c47cc7309e752eb087ecfbda7ff8a6e/bpf/bpf_lxc.c#L992-L1001) (Host-Ingress Policy Enforcement with HostFW + per-EP routes) can be bypassed? If it's no longer relevant, can it be removed?

Reply:
This was addressed by moving host firewall check above